src/lib/api.mli

```diff
@@ -125,8 +125,9 @@ module type BASIC_STRING = sig
   (** The type of the string. *)
 
   val max_string_length : int option
-  (** If the representation of strings is bounded,
-      the maximum length of a string. *)
+  (** If the representation of strings is bounded (by a constant,
+      something else that the process memory), the maximum length of a
+      string (assumed to be a {i number of [character]s}). *)
```

Comment:
"(by a constant other than the process memory), the maximum length of a string is assumed to be this number of characters"